### PR TITLE
Make the compile function of brotli version specific

### PR DESCRIPTION
**Pull request by Sinerum**
related se-sic/VaRA#492

### CODE REVIEW

File: varats/varats/projects/c_projects/brotli.py [hunk 1]
```diff
@@ -57,13 +83,25 @@ def run_tests(self) -> None:
...
+                "f9ab24a7aaee93d5932ba212e5e3d32e4306f748",
+                "5814438791fb2d4394b46e5682a96b68cd092803", ShortCommitHash
+            )
         c_compiler = bb.compiler.cc(self)
-        mkdir(brotli_version_source / "out")
-        with local.cwd(brotli_version_source / "out"):
-            with local.env(CC=str(c_compiler)):
-                bb.watch(local["../configure-cmake"])()
-            bb.watch(make)("-j", get_number_of_jobs(bb_cfg()))
+        if brotli_version in configure_revisions:
```
Comment (vulder) [inline varats/varats/projects/c_projects/brotli.py]:
```suggestion

        if brotli_version in configure_revisions:
```

**Review by vulder — CHANGES_REQUESTED** [review]
The compile function looks fine.
However, due to the revision blocking some tests are broken and need to be fixed before we can merge this.

File: varats/varats/projects/c_projects/brotli.py [hunk 2]
```diff
@@ -28,13 +33,21 @@ class Brotli(VProject):
...
-            remote="https://github.com/google/brotli.git",
-            local="brotli_git",
-            refspec="origin/HEAD",
-            limit=None,
-            shallow=False
+        block_revisions([
+            RevisionRange(
+                '8f30907d0f2ef354c2b31bdee340c2b11dda0fb0',
+                '378485b097fd7b80a5e404a3cb912f7b18f78cdb',
+                "building not unified"
```
Comment (boehmseb) [inline varats/varats/projects/c_projects/brotli.py]: What do you mean by "building not unified"? Maybe this message could be a bit more expressive.

Comment (Sinerum) [reply varats/varats/projects/c_projects/brotli.py]: There is no Makefile that builds everything. Every part of brotli so the encoder and decoder had seperate Makefiles at this point, but I just discoverd, that after a certain revision in this range the commanline tool was introduced, and the Makefile for this builds everything. This method has now also been added.

File: varats/varats/projects/c_projects/brotli.py [hunk 3]
```diff
@@ -57,13 +107,35 @@ def run_tests(self) -> None:
...
+        if brotli_version in simple_make_revisions:
+            with local.cwd(brotli_version_source / "tools"):
+                bb.watch(make)("-j", get_number_of_jobs(bb_cfg()))
+
+        elif brotli_version in configure_revisions:
+            with local.cwd(brotli_version_source):
+                with local.env(CC=str(c_compiler)):
+                    bb.watch(local["./configure"])()
+                bb.watch(make)("-j", get_number_of_jobs(bb_cfg()))
+
```
Comment (boehmseb) [inline varats/varats/projects/c_projects/brotli.py]:
```suggestion
```

File: varats/varats/projects/c_projects/brotli.py [hunk 4]
```diff
@@ -57,13 +107,35 @@ def run_tests(self) -> None:
...
-        mkdir(brotli_version_source / "out")
-        with local.cwd(brotli_version_source / "out"):
-            with local.env(CC=str(c_compiler)):
-                bb.watch(local["../configure-cmake"])()
-            bb.watch(make)("-j", get_number_of_jobs(bb_cfg()))
+
+        if brotli_version in simple_make_revisions:
+            with local.cwd(brotli_version_source / "tools"):
+                bb.watch(make)("-j", get_number_of_jobs(bb_cfg()))
+
```
Comment (boehmseb) [inline varats/varats/projects/c_projects/brotli.py]:
```suggestion
```

**Review by boehmseb — APPROVED** [review]

**Review by vulder — APPROVED** [review]

**Pull request merged**